customization-base/src/main/java/com/azure/autorest/customization/implementation/ls/EclipseLanguageServerFacade.java

```diff
@@ -93,22 +99,31 @@ private static Path getLanguageServerDirectory() throws IOException {
             downloadUrl = URI.create(DOWNLOAD_BASE_URL + "1.12.0/jdt-language-server-1.12.0-202206011637.tar.gz")
                 .toURL();
             languageServerPath = autorestLanguageServer.resolve("1.12.0");
-        } else {
+        } else if (javaVersion < 21) {
             // Eclipse JDT language server version 1.29.0 is the latest version that supports Java 17.
             // In the future this else statement may need to be replaced with an else if as newer versions of
             // Eclipse JDT language server may baseline on Java 21 (or later).
             downloadUrl = URI.create(DOWNLOAD_BASE_URL + "1.29.0/jdt-language-server-1.29.0-202310261436.tar.gz")
                 .toURL();
             languageServerPath = autorestLanguageServer.resolve("1.29.0");
+        } else {
+            // Eclipse JDT language server version 1.31.0 is the latest version that supports Java 21.
+            // In the future this else statement may need to be replaced with an else if as newer versions of
+            // Eclipse JDT language server may baseline on Java 25 (or later).
+            downloadUrl = URI.create(DOWNLOAD_BASE_URL + "1.31.0/jdt-language-server-1.31.0-202401111522.tar.gz")
+                .toURL();
+            languageServerPath = autorestLanguageServer.resolve("1.31.0");
         }
 
         Path languageServer = languageServerPath.resolve("jdt-language-server");
         if (!Files.exists(languageServerPath) || !Files.exists(languageServer)) {
             Files.createDirectories(languageServerPath);
             Path zipPath = languageServerPath.resolve("jdt-language-server.tar.gz");
+            logger.info("Downloading Eclipse JDT language server from {} to {}", downloadUrl, zipPath);
```

Comment:
It's also worth adding a log when we skip the download because we found a downloaded local copy already. Log the path to the local copy in case we want to just delete and start from scratch.

Reply:
this is already being done on line 78

`logger.info("Starting Eclipse JDT language server at {}", languageServerPath);`